### DISCUSSION TITLE
[expo-custom-dev-client] [iOS] Expo Custom Dev Client crashing to SplashScreen without any error logging, Native/Release Variant doesn't crash. How to find the root cause?

### DISCUSSION THREAD

**Discussion by junaiddarajat** [expo/expo#17552] [Archived: Questions]
Using Expo SDK 44, and Expo Custom Dev Client, I've forked react-native-incallmanager and have made some amends, more specifically stuff from Jitsi Meet. You can check my fork from my profile.

The question I want to ask is; when using Expo Custom Dev Client; is there a way we can find any sort of logging or indication to why the application is crashing? Normally, I'd just make release variant and check from XCode debugger / console and see any native errors, but I'm getting nothing as the crash doesn't happen on the release variant BUT only on Custom Dev Client.

The crash is as follows, I just see the Splash Screen and there are no logs whatsoever. So what are the ways I can attain some sort of logging? Do note that I've followed the debugging guidelines and already import the custom dev client before App.js/jsx.

Here's the code which when calling it crashes the app:

```
    import { NativeEventEmitter, NativeModules } from "react-native";

    const { InCallManager } = NativeModules;

    useEffect(() => {
        const incallManagerEmitter = new NativeEventEmitter(InCallManager);
        const subscription = incallManagerEmitter.addListener("AudioOutputDevices", (data) => console.log(data));

        console.log("CALLING getAudioOutputDevices(): ");
        InCallManager.getAudioOutputDevices();
        return () => subscription.remove();
    }, []);
```

Now, whenever I make a release variant by either: Launching XCode and Editing Scheme as "Release", or running:
`expo run:ios --configuration release`; it doesn't crash.

I want to find the root cause of the crash, so any guidance is highly appreciated. Thanks so much for your time and consideration!

**Comment by lukmccall** [comment]
Hi @junaiddarajat, you can do a couple of things to access logs:
1. Run your app directly from the XCode: 
- run `expo prebuild` 
- open generated project (under `iOS` directory - `*.xcworkspace` file) in XCode
2. You can follow our guide: https://docs.expo.dev/workflow/logging/#view-logs-for-an-ios-simulator

**Comment by StevenBlack** [comment]
Seeing the very similar things here on both SDK 44 and on SDK 45, developing with iOS and Android developer devices, both phones and tablets.

Here is what we see:

* Expo Go execution on iOS and Android **runs exactly once**. Upon second and subsequent execution, quiet crash at the first screen with no screen contents showing. (Note: the splash shows fine. In our case, the first screen is a login screen.)

* <s>We have has this problem since the May 15 2022 release of Expo Go.</s>

* **Deleting and reinstalling Expo Go** in iOS and Android devices we get the **same thing**: executes fine, exactly once.

* We have tried this `[remove Expo Go, reinstall Expo Go]` cycle several times. Same thing, regardless of device type.  Expo Go works exactly once, then the only way we've had a subsequent run is to delete, and reinstall, Expo Go.

**THEREFORE**
* Expo Go, May 15 2022 release, is broken.

**Reply by brentvatne** [reply]
@StevenBlack - we can't reproduce this. can you share your logs from connecting your device and running `adb logcat`?

**Comment by StevenBlack** [comment]
@brentvatne thank you for this interesting learning opportunity.

Very noob to this facet of Android, here's what I did, and what I got.

1) Download, install, and config Android Studio on MacOS.
2) Seven-tap this Galaxy A21.
3) Run `./adb logcat -v threadtime [DeviceId] > /tmp/android-debug.log`
4) `expo start`, shot the QR code, ran the app to crash, which isn't long at all (first screen, just beyond splash screen).

File `/tmp/android-debug.log` contains 267,417 lines. How do you want it?

Alternarely instruct me on how to get you a more terse log file?

**Comment by StevenBlack** [comment]
@brentvatne something odd.  I didn't `Ctrl + C` this to stop it. Logging stopped on its own. The time stamps in the log file make sense, and concord.

**Comment by StevenBlack** [comment]
@brentvatne some more information. I have an iPad here with a pre-May 7 release of Expo Go. Automatic updates turned off on this iPad.  This is SDK 44 and we are seeing the same issue: Expo Go runs exactly once.

Therefore it's something in our project that's triggering this behavior in Expo Go. This behavior has been in Expo Go since pre May 7th.
